server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java

```diff
@@ -89,26 +89,36 @@
 
     private final Collection<ClusterStateListener> clusterStateListeners = new CopyOnWriteArrayList<>();
     private final Collection<TimeoutClusterStateListener> timeoutClusterStateListeners =
-        Collections.newSetFromMap(new ConcurrentHashMap<TimeoutClusterStateListener, Boolean>());
+        Collections.newSetFromMap(new ConcurrentHashMap<>());
 
     private final LocalNodeMasterListeners localNodeMasterListeners;
 
     private final Queue<NotifyTimeout> onGoingTimeouts = ConcurrentCollections.newQueue();
 
     private final AtomicReference<ClusterState> state; // last applied state
 
-    private final String nodeName;
+    private final Supplier<PrioritizedEsThreadPoolExecutor> threadPoolExecutorFactory;
 
     private NodeConnectionsService nodeConnectionsService;
 
     public ClusterApplierService(String nodeName, Settings settings, ClusterSettings clusterSettings, ThreadPool threadPool) {
+        this(settings, clusterSettings, threadPool,
+            () -> EsExecutors.newSinglePrioritizing(
+                nodeName + "/" + CLUSTER_UPDATE_THREAD_NAME,
+                daemonThreadFactory(nodeName, CLUSTER_UPDATE_THREAD_NAME),
+                threadPool.getThreadContext(),
+                threadPool.scheduler()));
+    }
+
+    public ClusterApplierService(Settings settings, ClusterSettings clusterSettings, ThreadPool threadPool,
+        Supplier<PrioritizedEsThreadPoolExecutor> threadPoolExecutorFactory) {
```

Comment:
In MasterService, this was mocked out by having a protected method `createThreadPoolExecutor()`. Can you align it so that both classes allow the mocking in the same way?